mocks/mocks.go

```diff
@@ -516,3 +495,47 @@ func (m *Mailer) Close() error {
 func (m *Mailer) Connect() error {
 	return nil
 }
+
+// mockSAWithFailedChallenges is a mocks.StorageAuthority that has
+// a `GetAuthorization` implementation that can return authorizations with
+// failed challenges.
+type SAWithFailedChallenges struct {
```

Comment:
Could you move this into wfe_test.go, and downcase the type name and the `Clk` member? Relatedly, does this mock need a clock?

Reply:
I mentioned why I put the mock there here: https://github.com/letsencrypt/boulder/pull/3039#discussion_r137108289 

It needs a clock only to set the expires time. If an authz has a nil expires time other code breaks.

Reply:
Aha, I missed that; thanks for pointing it out.